packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1921,11 +2009,17 @@ export class SnapController extends BaseController<
     }
 
     if (!encrypted) {
-      // For performance reasons, we do not validate that the state is JSON, since we control serialization.
-      return JSON.parse(state);
+      // For performance reasons, we do not validate that the state is JSON,
+      // since we control serialization.
+      const json = JSON.parse(state);
+      runtime.unencryptedState = json;
+
+      return json;
     }
 
     const decrypted = await this.#decryptSnapState(snapId, state);
+    runtime.state = decrypted;
```

Comment:
We should be clearing these again when the MetaMask client is locked and the vault is unloaded from memory.

Reply:
Done!